### PR TITLE
User orders show and index pages

### PR DESCRIPTION
**Pull request by CoopTang**
- (#18) Add profile orders index page to view all a user's orders.
- (#17) Add profile orders show page to view a single order by a user. Also added sad path for users that try to access orders that don't exist or don't belong to the current user.

### CODE REVIEW

File: app/views/user_orders/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<h1>ID: <%= @order.id %></h1>
+<article id='order-info'>
+  <%= render partial: 'partials/order_info',
+    locals: {
+      id: @order.id,
+      created_at: @order.created_at,
+      updated_at: @order.updated_at,
+      status: @order.status,
+      item_orders: @order.item_orders,
```
Comment (freeheeling) [inline app/views/user_orders/show.html.erb:9]: ~~Does each item ordered include a thumbnail of the item, per the user story?~~ Never mind, I see you included this in your test.

File: config/routes.rb [hunk 2]
```diff
@@ -26,9 +26,11 @@
   get '/profile', to: 'users#show'
   get '/profile/edit', to: 'users#edit_profile'
   get '/profile/edit_password', to: 'users#edit_password'
-  get '/profile/orders', to: 'users#show' # Change this when we come up with a solution
   patch '/profile/update', to: 'users#update_profile'
   patch '/profile/update_password', to: 'users#update_password'
+  
+  get '/profile/orders', to: 'user_orders#index'
+  get '/profile/orders/:id', to: 'user_orders#show', as: 'profile_order'
```
Comment (freeheeling) [inline config/routes.rb:33]: Solved!

File: spec/features/user_orders/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        expect(page).to have_content("Subtotal: #{item_order_2.subtotal}")
+      end
+    end
+
+    it 'renders a 404 if the show page does not exist' do
+      meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80_203)
+      brian = Merchant.create(name: "Brian's Dog Shop", address: '125 Doggo St.', city: 'Denver', state: 'CO', zip: 80_210)
+
+      tire = meg.items.create(name: 'Gatorskins', description: "They'll never pop!", price: 100, image: 'https://www.rei.com/media/4e1f5b05-27ef-4267-bb9a-14e35935f218?size=784x588', inventory: 12)
+      pull_toy = brian.items.create(name: 'Pull Toy', description: 'Great pull toy!', price: 10, image: 'http://lovencaretoys.com/image/cache/dog/tug-toy-dog-pull-9010_2-800x800.jpg', inventory: 32)
```
Comment (freeheeling) [inline spec/features/user_orders/show_spec.rb:64]: Is it necessary to include merchants and items in this part of the test? If so, could the identical setup in both it blocks be abstracted to a before(:each)?

File: app/views/user_orders/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+<% current_user.orders.each do |order| %>
+  <article id='order-<%= order.id %>'>
+  <p>ID: <%= link_to "#{order.id}", profile_order_path(order) %></p>
+    <%= render partial: 'partials/order_info',
+      locals: {
+        id: order.id,
+        created_at: order.created_at,
+        updated_at: order.updated_at,
+        status: order.status,
+        item_orders: nil,
```
Comment (freeheeling) [inline app/views/user_orders/index.html.erb:11]: Could the item_orders local also be omitted if it's not rendered, or must it be included?

**Review by freeheeling — COMMENTED** [review]
Looks great overall! Just a few questions to better improve my understanding.

**Review by cjkelling — APPROVED** [review]
Great job on testing this new functionality and thinking of potential sad paths. Everything appears to be functioning how the user story describes.

**Pull request merged**